File: Data Structures & Algorithms/binary-tree-diameter/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public int DiameterOfBinaryTree(TreeNode root) {
        if(root == null) return 0;
        int diameter = depth(root.left) + depth(root.right);
        int diameterOfLeft = DiameterOfBinaryTree(root.left);
        int diameterOfRight = DiameterOfBinaryTree(root.right);

        return Math.Max(diameter, Math.Max(diameterOfLeft, diameterOfRight));

    }


    public int depth(TreeNode root){
        if(root == null) return 0;
        return 1 + Math.Max(depth(root.left),depth(root.right)); 
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int goodNodes(TreeNode root) {
        int maxVal = Integer.MIN_VALUE;
        return solve(root, maxVal);
        
    }

    public int solve(TreeNode root, int maxVal){
        if(root == null) return 0;

        if(root.val >= maxVal){
            return 1 + solve(root.left, root.val) + solve(root.right, root.val);
        }
        else{
            return solve(root.left, maxVal) + solve(root.right, maxVal);
        }
    }
}
